src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -278,6 +276,7 @@ public ImmutableArray<int> ToImmutableArray()
                 if (seenNamedParams)
                 {
                     // Unnamed arguments after a named argument corresponding to a params parameter cannot correspond to any parameters
+                    Debug.Assert(expanded);
```

Comment:
So, going though this code, I couldn't convince myself that this entire `if` was necessary, either before or after the change. I checkout out this PR, commented out the `return null;`, and ran all the compiler tests. There were no failures. Should we consider whether this entire bit of logic needs to be simplified or removed?

Reply:
My goal was to get rid of the ```IsValidParams``` check. Therefore, the suggested level of investigation is outside of scope of this PR. Feel free to open a follow up issue. It is quite possible we are missing a test. A possible scenario that could be affected is named params argument followed by an unnamed argument. It is quite possible that we also detect similar situation elsewhere and removal of logic won't be observable overall. However, I think the result of the method itself is likely to be affected.

Reply:
> A possible scenario that could be affected is named params argument followed by an unnamed argument.

Right, that was my first suspicion as well, but that will hit the `if` just below and have the same result; if the `params` parameter was named, we will have already decided that it was a valid `params` parameter, so it _must_ be used out of position. We'll then return `null` for the unnamed parameter, which isn't observably different. I can open a followup issue.

Reply:
https://github.com/dotnet/roslyn/issues/70923

Reply:
> so it must be used out of position

Why is that? It could be the only parameter

Reply:
In either of those cases, we wouldn't get here. If it's used in position, it must be the last argument, and there are no more arguments to process. Same for being the only argument.

Reply:
>

For this scenario we are getting into this `if` when ```seenOutOfPositionNamedArgument``` is `false`:
```
class Program
{
    static void Main()
    {
        Test(a: 1, 2);
    }

    static void Test(params int[] a)
    {
    }
}
``` 

If I move the instruction pointer out of the `if`, it looks like we fail to report an error that is expected in this scenario